airflow/api_connexion/endpoints/connection_endpoint.py

```diff
@@ -129,3 +133,21 @@ def post_connection(session):
         session.commit()
         return connection_schema.dump(connection)
     raise AlreadyExists(detail=f"Connection already exist. ID: {conn_id}")
+
+
+@security.requires_access([(permissions.ACTION_CAN_CREATE, permissions.RESOURCE_CONNECTION)])
+def test_connection():
+    """Test a connection"""
+    body = request.json
+    try:
+        data = connection_schema.load(body)
+        dummy_conn_id = get_random_string()
+        data['conn_id'] = dummy_conn_id
+        conn = Connection(**data)
+        conn_env_var = f'AIRFLOW_CONN_{dummy_conn_id.upper()}'
+        os.environ[conn_env_var] = conn.get_uri()
```

Comment:
Why do we need to do this? We've already got the connection object created...

Reply:
Yes, but some hooks tried to get the conn object from their __init__, and as we are not storing anything in DB, they fail. Setting it in env let them use the connection.

Reply:
Oh yes, gotcha. Could you add a comment saying why we need this please